src/frontend/src/handler/create_index.rs

```diff
@@ -247,7 +269,15 @@ fn assemble_materialize(
 
     PlanRoot::new(
         logical_project,
-        RequiredDist::AnyShard,
+        if distributed_by_columns.is_empty() {
+            // We use the whole index columns as distributed key by default if users
+            // haven't specify the distributed by columns.
+            RequiredDist::PhysicalDist(Distribution::HashShard((0..index_columns.len()).collect()))
+        } else {
+            RequiredDist::PhysicalDist(Distribution::HashShard(
+                (0..distributed_by_columns.len()).collect(),
```

Comment:
This seems to depend on the `index_columns.starts_with(&distributed_by_columns)` check above, and only the length info is useful after the check. Shall we just pass `distributed_by_columns_len` (alongside `index_columns`) from the caller?

Reply:
Sure.